### PR TITLE
OCT-628 - Viewing other author's affiliations - UI

### PR DESCRIPTION
**Pull request by shaunajisc**
The purpose of this PR was to display all author's affiliations on the draft/ locked publication page within the acceptance tracker.

This PR contains: 
-  The addition of a Simple Modal component - this can be used to display info and can be dismissed by clicking the cancel button or clicking outside of the modal
-  The addition of a column in the acceptance tracker that displays each user's affiliations 
-  Either an eye button or edit button within that row that allows users to see all of the affiliations  
-  An Affiliations Card (this was created by Florin and is in one of his PRs - I'll fix the conflicts once his is merged in) 

---

### Acceptance Criteria:

As per [OCT-628](https://jiscdev.atlassian.net/jira/software/projects/OCT/boards/836?selectedIssue=OCT-628)

---

### Screenshots:

https://github.com/JiscSD/octopus/assets/100135505/828db842-87c0-4430-9d2b-874bc1c95d96

[OCT-628]: https://jiscdev.atlassian.net/browse/OCT-628?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: ui/src/components/SimpleModal/index.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+                                <div className="mt-3 sm:mt-5">
+                                    <HeadlessUI.Dialog.Title
+                                        as="h3"
+                                        className="text-center font-montserrat text-lg font-semibold leading-6 text-grey-900"
+                                    >
+                                        {props.title}
+                                    </HeadlessUI.Dialog.Title>
+                                    <h4 className=" eading-6 mt-6 font-montserrat text-base font-semibold text-grey-900">
+                                        {props.subTitle}
+                                    </h4>
```
Comment (finlay-jisc) [inline ui/src/components/SimpleModal/index.tsx:61]: Should we check if we have props.subTitle before rendering this h4, since it's optional?

**Review by finlay-jisc — COMMENTED** [review]
Suggested a few adjustments. One other thing I thought of was: maybe the modal should have a cross in the top right or something, so that if there are lots of affiliations the modal can be closed without scrolling to the bottom.

File: ui/src/components/ApprovalsTracker/index.tsx [hunk 2]
```diff
@@ -272,26 +326,41 @@ const ApprovalsTracker: React.FC<Props> = (props): React.ReactElement => {
...
                         ref={authorEmailRef}
                     />
                     {authorEmailError && <Components.Alert severity="ERROR" title={authorEmailError} />}
                 </Components.Modal>
             )}
+            <Components.SimpleModal
+                open={Boolean(selectedAuthorAffiliations)}
+                negativeActionCallback={() => setSelectedAuthorAffiliations(null)}
+                cancelButtonText="Close"
+                title={`${selectedAuthorAffiliations?.user?.firstName} ${selectedAuthorAffiliations?.user?.lastName}'s Affiliation(s) for this publication`}
```
[the remark this answers is not in the document]
Comment (shaunajisc) [reply ui/src/components/ApprovalsTracker/index.tsx:353]: Good spot - thanks!

**Review by finlay-jisc — DISMISSED** [review]

File: ui/src/components/ApprovalsTracker/index.tsx [hunk 3]
```diff
@@ -146,6 +149,19 @@ const ApprovalsTracker: React.FC<Props> = (props): React.ReactElement => {
         [props.publication.createdBy, user?.id]
     );
 
+    const displayArrayOfAffiliations = (affiliations: Interfaces.MappedOrcidAffiliation[]) => {
```
Comment (florin-holhos) [inline ui/src/components/ApprovalsTracker/index.tsx:152]: whenever displaying these affiliations, they need to be sorted. I have created a helper function for this in /lib/helpers.tsx.
image

**Comment by Harry-Jisc** [comment]
I'm not able to view the affiliations of the corresponding author as another author of the publication, only the first 2/3 depending if there is more. 

0d756b8c71fdbd9259f99c70e5523508

**Comment by shaunajisc** [comment]
> I'm not able to view the affiliations of the corresponding author as another author of the publication, only the first 2/3 depending if there is more.
> 
> 0d756b8c71fdbd9259f99c70e5523508

Thanks for spotting that - I've pushed a commit that should resolve that bug

**Review by Harry-Jisc — APPROVED** [review]

**Pull request merged**